src/tokenizer/buffer_queue.rs

```diff
@@ -20,16 +21,16 @@ pub use self::SetResult::{FromSet, NotFromSet};
 
 struct Buffer {
     /// Byte position within the buffer.
-    pub pos: usize,
+    pub pos: u32,
     /// The buffer.
-    pub buf: String,
+    pub buf: Tendril,
```

Comment:
But the owned variant of a `Tendril` doesn't.

Eventually I would like to replace `BufferQueue` with a rope, as discussed elsewhere.